proposals/infra/INF-0004-inproc-outofproc-compiler-library-support.md

```diff
@@ -0,0 +1,138 @@
+<!-- {% raw %} -->
+
+# In-process only or In-process and Out-of-process support for the compiler library
+
+* Proposal: [0004](INF-0004-inproc-outofproc-compiler-library-support.md)
+* Author(s): [Cooper Partin](https://github.com/coopp)
+* Sponsor: [Cooper Partin](https://github.com/coopp)
+* Status: **Under Consideration**
+* Impacted Project(s): (Clang)
+
+* Issues:
+
+## Introduction
+
+An effort is underway to bring compilation support into clang for HLSL based
+shaders.  A C style library will be built to enable applications to compile
+a shader from their own processes in addition to being able to launching the
+clang process.
+
+There are two proposed ways one could consume the shader compiler library and
+compile a shader today.
+
+* **In Process** - An application links against the shader compiler library to
+bring in support for shader compilation as a static library
+(full implementation) or a dynamic library. Compilation is performed within the
+calling application's process.
+
+The DirectX Shader library does not currently support being linked as a
+static library.  It is a COM interface library that brings in compilation
+support dynamically. Both static and dynamic in-proc solutions will be 
+supported in the HLSL clang based library.
+
+* **Out of Process** - An application links against the shader compiler library and
+uses the apis to compile which create one or more processes on the
+application's behalf to compile one or more shaders. Compilation is performed
+outside the calling application's process and isolated away to a worker
+process.
+
+In-Process must be supported because it is already a method that applications
+use today to consume the DirectX Shader Compiler library.
```

Comment:
> In-Process must be supported because it is already a method that applications use today to consume the DirectX Shader Compiler library.

I disagree with this logic.  Just because it is supported by DXC does not mean that this is the right solution for clang.

Maybe we'll still decide that we need to support in-proc, but if we do I think we need something stronger than "this is what DXC did" as a reason to do it.

Reply:
I removed the leaning on DXC's past as evidence to support a direction.  It should be top of mind for features, but for architecture we can be more creative as to what we want the new clang based architecture to be.